diff --git a/Makefile b/Makefile
index d60e69a47..b40b52826 100644
--- a/Makefile
+++ b/Makefile
@@ -48,8 +48,8 @@ E2E_HELM_OPTIONS ?= --set image.smb.repository=$(REGISTRY)/$(IMAGENAME) --set im
 E2E_HELM_OPTIONS += ${EXTRA_HELM_OPTIONS}
 # Generate all combination of all OS, ARCH, and OSVERSIONS for iteration
 ALL_OS = linux windows
-ALL_ARCH.linux = arm64 amd64 ppc64le
-ALL_OS_ARCH.linux = linux-arm64 linux-arm-v7 linux-amd64 linux-ppc64le
+ALL_ARCH.linux = arm64 riscv64 amd64 ppc64le
+ALL_OS_ARCH.linux = linux-arm64 linux-riscv64 linux-arm-v7 linux-amd64 linux-ppc64le
 ALL_ARCH.windows = amd64
 ALL_OSVERSIONS.windows := 1809 ltsc2022
 ALL_OS_ARCH.windows = $(foreach arch, $(ALL_ARCH.windows), $(foreach osversion, ${ALL_OSVERSIONS.windows}, windows-${osversion}-${arch}))
@@ -153,13 +153,13 @@ container: smb
 container-linux:
 	docker buildx build --no-cache --pull --output=type=$(OUTPUT_TYPE) --platform="linux/$(ARCH)" \
 		--provenance=false --sbom=false \
-		-t $(IMAGE_TAG)-linux-$(ARCH) --build-arg ARCH=$(ARCH) -f ./cmd/smbplugin/Dockerfile .
+		-t $(IMAGE_TAG)-linux-$(ARCH) --build-arg DOCKER_ARCH=$(ARCH) --build-arg ARCH=$(ARCH) -f ./cmd/smbplugin/Dockerfile .
 
 .PHONY: container-linux-armv7
 container-linux-armv7:
 	docker buildx build --no-cache --pull --output=type=$(OUTPUT_TYPE) --platform="linux/arm/v7" \
 		--provenance=false --sbom=false \
-		-t $(IMAGE_TAG)-linux-arm-v7 --build-arg ARCH=arm/v7 -f ./cmd/smbplugin/Dockerfile .
+		-t $(IMAGE_TAG)-linux-arm-v7 --build-arg DOCKER_ARCH=armv7 --build-arg ARCH=arm/v7 -f ./cmd/smbplugin/Dockerfile .
 
 .PHONY: container-windows
 container-windows:
